tools/PI/DevHome.PI/Controls/EditToolsControl.xaml.cs

```diff
@@ -12,16 +12,37 @@ public sealed partial class EditToolsControl : UserControl
     public EditToolsControl()
     {
         InitializeComponent();
-        ToolsDataGrid.ItemsSource = ExternalToolsHelper.Instance.AllExternalTools;
+        EnableUnregisterButton();
     }
 
-    private void DeleteToolButton_Click(object sender, RoutedEventArgs e)
+    private void UnregisterToolButton_Click(object sender, RoutedEventArgs e)
     {
-        var selectedItem = ToolsDataGrid.SelectedItem;
+        var selectedItems = ToolsDataGrid.SelectedItems;
+        for (var i = selectedItems.Count - 1; i >= 0; i--)
+        {
+            if (selectedItems[i] is ExternalTool tool)
+            {
+                ExternalToolsHelper.Instance.RemoveExternalTool(tool);
+            }
+        }
 
-        if (selectedItem is ExternalTool)
+        EnableUnregisterButton();
+    }
+
+    private void EnableUnregisterButton()
+    {
+        if (ToolsDataGrid.SelectedItems is not null && ToolsDataGrid.SelectedItems.Count > 0)
```

Comment:
NIT: `UnregisterToolButton.Visibility = ToolsDataGrid.SelectedItems?.Count > 0;`

Reply:
Thanks, fixed in subsequent commit.